### PR TITLE
Upgrade to TypeScript 4.4

### PR DESCRIPTION
**Pull request by roluk**
[TypeScript 4.4 release notes](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/).

* Catch variables have `unknown` type by default.
    * Handle iModel opening and closing exceptions more carefully to avoid crashes.

### CODE REVIEW

File: app/frontend/src/app/ITwinJsApp/InitializedApp.tsx [hunk 1]
```diff
@@ -206,7 +209,10 @@ function useIModel(backendApi: BackendApi, path: string): IModelConnection | und
               await openedIModel.close();
             }
           } catch (error) {
-            displayErrorToast(IModelApp.i18n.translate("App:error:imodel-close", { imodel: path }), error.message);
+            displayErrorToast(
+              IModelApp.i18n.translate("App:error:imodel-close", { imodel: path }),
+              (error && typeof error === "object") ? (error as { message: unknown }).message : undefined,
```
Comment (grigasp) [inline app/frontend/src/app/ITwinJsApp/InitializedApp.tsx:214]: wouldn't it be more convenient to have this check in `displayErrorToast` function and pass the `error` object as `unknown` to it?

Comment (roluk) [reply app/frontend/src/app/ITwinJsApp/InitializedApp.tsx:214]: I would prefer to keep `displayErrorToast` function

Comment (roluk) [reply app/frontend/src/app/ITwinJsApp/InitializedApp.tsx]: `displayErrorToast` may be used in other contexts too, so I want to keep it generic. Hope I addressed your issue in a494560955e652f2ab36cbd0bc962fcd5a4c259b.

**Review by grigasp — APPROVED** [review]

File: app/frontend/src/app/ITwinJsApp/InitializedApp.tsx (hunk 1 above)
Comment (grigasp) [reply app/frontend/src/app/ITwinJsApp/InitializedApp.tsx]: yes, thanks

**Pull request merged**